stdlib/http/src/main/java/org/ballerinalang/net/http/websocket/client/WebSocketClientListener.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http:www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http:www.apache.orglicensesLICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.net.http.websocket.client;
+
+import org.ballerinalang.jvm.values.ObjectValue;
+import org.ballerinalang.net.http.websocket.WebSocketConstants;
+import org.ballerinalang.net.http.websocket.WebSocketResourceDispatcher;
+import org.ballerinalang.net.http.websocket.server.WebSocketOpenConnectionInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.transport.http.netty.contract.websocket.WebSocketCloseMessage;
+import org.wso2.transport.http.netty.contract.websocket.WebSocketConnection;
+
+import java.io.IOException;
+
+import static org.ballerinalang.net.http.websocket.WebSocketConstants.STATEMENT;
+import static org.ballerinalang.net.http.websocket.WebSocketConstants.STATUS_CODE_ABNORMAL_CLOSURE;
+import static org.ballerinalang.net.http.websocket.WebSocketUtil.handleExceptionAndDispatchCloseMessage;
+import static org.ballerinalang.net.http.websocket.WebSocketUtil.hasRetryConfig;
+import static org.ballerinalang.net.http.websocket.WebSocketUtil.reconnect;
```

Comment:
Shall we please not use static imports unless it really makes sense: https://stackoverflow.com/a/162195/3599535